tests/jobs/test_scheduler_job.py

```diff
@@ -207,7 +207,7 @@ def test_process_executor_events(self, mock_stats_incr, mock_task_callback, dag_
 
         self.scheduler_job._process_executor_events(session=session)
         ti1.refresh_from_db()
-        assert ti1.state == State.FAILED
+        assert ti1.state == State.QUEUED
         mock_task_callback.assert_called_once_with(
```

Comment:
Hmmm..  I wonder if we can just handle failure in _process_executor_events itself if there are no callbacks. If there are no callbacks we probably don't need to send it to dag parsing process 

And we need a comment here, to say the state will remain in Queue here and will be set to failed in dag parsing process.

Reply:
Ok. But I think we may miss the logs?

Reply:
If there are no callbacks what logs did you expect?

Oh, and sending it to the dag file parser process still won't produce any logs that in the "right" location to get picked up by the webserver.

Reply:
The issue I face now is that to be able to check if there are still retries, we need the task object. `ti.is_eligible_to_retry` requires that `ti.task` exists